# One-Sample T-Test

-   null hypothesis (H0): the mean of the areas is 5000.
-   alternative hypothesis (H1): the mean of the areas is not 5000.


In [2]:
import pandas as pd

data = pd.read_csv("./data/areas.csv")
data

area
0   1000
1   1500
2   2300
3   3540
4   4120
5   4560
6   5490
7   3460
8   4750
9   2300
10  9000
11  8600
12  7100

In [3]:
import scipy.stats as stats

_, p_value = stats.ttest_1samp(a=data, popmean=5000)
p_value

array([0.44346471])

In [ ]:
p_value > 0.05

array([ True])

The p-value is bigger than the alpha value, so H0 is accepted and H1 is rejected

In [5]:
data.mean()

area    4440.0
dtype: float64

This can be confirmed as the actual mean is 4440, which is not far off

In [8]:
_, p_value = stats.ttest_1samp(a=data.subtract(1000), popmean=5000)
p_value > 0.05

array([False])

In [9]:
data.subtract(1000).mean()

area    3440.0
dtype: float64

Here, the mean is 1000 smaller, so H0 is rejected and H1 is accepted